### PR TITLE
Add recipe for codechecker

### PR DESCRIPTION
**Pull request by mwittgen**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/codechecker/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    - gitpython
+    - lxml
+    - mypy_extensions
+    - pip
+    - portalocker
+    - psutil
+    - python
+    - pyyaml
+    - sqlalchemy
+    - thrift
```
Comment (dopplershift) [inline recipes/codechecker/meta.yaml:37]: Looks like a copy/paste went wrong and the requirements got duplicate?
```suggestion
```

Comment (mwittgen) [reply recipes/codechecker/meta.yaml:37]: This was generated more or less from conda skeleton. Only the order was wrong. Suggested change does not work.

File: recipes/codechecker/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
```
Comment (dopplershift) [inline recipes/codechecker/meta.yaml:37]: I don't think most of these are needed at *build* time. Also, Python should be listed under `host`:
```suggestion
    - {{ compiler('cxx') }}
  host:
    - python
    - pip
```

File: recipes/codechecker/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    - merge-clang-extdef-mappings = codechecker_merge_clang_extdef_mappings.cli:main
+    - post-process-stats = codechecker_statistics_collector.cli:main
+    - report-converter = codechecker_report_converter.cli:main
+    - tu_collector = tu_collector.tu_collector:main
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - python
+    - {{ compiler('cxx') }}
```
Comment (dopplershift) [inline recipes/codechecker/meta.yaml:25]: Only need `python` listed under the `host` section:
```suggestion
  build:
    - {{ compiler('cxx') }}
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
Just a minor cleanup and this is ready.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**